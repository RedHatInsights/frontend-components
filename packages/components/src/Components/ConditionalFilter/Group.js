import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { Select, SelectOption, SelectVariant, SelectGroup, Radio, Checkbox } from '@patternfly/react-core';
import Text from './Text';
import isEqual from 'lodash/isEqual';

export const groupType = {
    checkbox: 'checkbox',
    radio: 'radio',
    plain: 'plain'
};

class Group extends Component {
    state = {
        isExpanded: false,
        selected: {},
        filterBy: /./ // Use Regex here to ignore case match, set to all characters
    }

    onToggle = isExpanded => {
        this.setState({
            isExpanded,
            filterBy: /./
        });
    };

    componentDidUpdate({ selected: prevSelected }) {
        const { selected } = this.props;
        if (!isEqual(prevSelected, selected)) {
            this.setState({
                selected
            });
        }
    }

    mapItems = ({ groupValue, onSelect, groupLabel, groupId, type, items, ...group }, groupKey) => {
        return items.filter(item =>
            (groupValue && this.state.filterBy.test(groupValue)) ||
            (groupLabel && this.state.filterBy.test(groupLabel)) ||
            (item.value && this.state.filterBy.test(item.value)) ||
            (item.label && this.state.filterBy.test(item.label))
        ).map(({ value, isChecked, onClick, label, props: itemProps, id, ...item }, key) => (
            <SelectOption
                {...item}
                key={id || key}
                value={String(value || id || key)}
                onClick={e => {
                    if (e.target.tagName === 'LABEL') {
                        e.preventDefault();
                        e.stopPropagation();
                    }

                    const clickedGroup = {
                        value: groupValue,
                        label: groupLabel,
                        id: groupId,
                        type,
                        items,
                        ...group
                    };
                    const clickedItem = { value, label, id, type, ...group };
                    const props = [
                        e,
                        clickedGroup,
                        clickedItem,
                        groupValue || groupKey,
                        value || key
                    ];

                    this.onSelect(...props);
                    onSelect && onSelect(...props);
                    onClick && onClick(...props);
                }}
            >
                {
                    type === groupType.checkbox &&
                    <Checkbox
                        {...itemProps}
                        label={label}
                        isChecked={
                            isChecked ||
                            this.isChecked(groupValue || groupKey, value || key) ||
                            false
                        }
                        onChange={(value, event) => {
                            item.onChange && item.onChange(value, event);
                        }}
                        name={item.name || value || `${groupKey}-${key}`}
                        id={id || value || `${groupKey}-${key}`}
                    />
                }
                {
                    type === groupType.radio &&
                    <Radio
                        isChecked={
                            isChecked ||
                            this.isChecked(groupValue || groupKey, value || key) ||
                            false
                        }
                        onChange={(value, event) => {
                            item.onChange && item.onChange(value, event);
                        }}
                        value={value || key}
                        name={item.name || value || `${groupKey}-${key}`}
                        label={label}
                        id={id || value || `${groupKey}-${key}`}
                    />
                }
                {(type !== groupType.checkbox && type !== groupType.radio) ? label : ''}
            </SelectOption>
        ));
    }

    calculateSelected = ({ type }, groupKey, itemKey) => {
        const { selected } = this.state;
        const { selected: propSelected } = this.props;
        const activeGroup = selected[groupKey] || propSelected[groupKey];
        if (activeGroup) {
            if (type !== groupType.radio && activeGroup[itemKey]) {
                return {
                    ...propSelected,
                    ...selected,
                    [groupKey]: {
                        ...activeGroup || {},
                        [itemKey]: false
                    }
                };
            }

            return {
                ...propSelected,
                ...selected,
                [groupKey]: {
                    ...type !== groupType.radio ? (activeGroup || {}) : {},
                    [itemKey]: true
                }
            };
        }

        return {
            ...propSelected,
            ...selected,
            [groupKey]: {
                [itemKey]: true
            }
        };
    }

    onSelect = (event, group, item, groupKey, itemKey) => {
        let newSelection = this.calculateSelected(group, groupKey, itemKey);

        const { onChange } = this.props;

        if (onChange) {
            onChange(event, newSelection, group, item);
            this.setState({ selected: {} });
        }

        this.setState({
            selected: newSelection,
            filterBy: /./
        });
    };

    isChecked = (groupValue, itemValue) => {
        const { selected: stateSelected } = this.state;
        const { selected: propSelected } = this.props;
        const selected = {
            ...propSelected,
            ...stateSelected
        };
        if (typeof selected[groupValue] === 'undefined') {
            return false;
        }

        return Boolean(selected[groupValue][itemValue]);
    }

    customFilter = (e) => {
        let input;

        try {
            input = new RegExp(e.target.value, 'i');
        } catch (err) {
            input = new RegExp(e.target.value.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'), 'i');
        }

        this.setState({ filterBy: input });
    }

    render() {
        const { isExpanded } = this.state;
        const { groups, items, placeholder, className, selected, isFilterable, onFilter } = this.props;

        const filterItems = items || groups;

        return (<Fragment>
            { !filterItems || (filterItems && filterItems.length <= 0) ? <Text { ...this.props } value={ `${selected}` } /> : <Select
                className={ className }
                variant={ (isFilterable || onFilter) ? SelectVariant.typeahead : SelectVariant.single }
                aria-label="Select Input"
                onToggle={ this.onToggle }
                isExpanded={ isExpanded }
                onSelect={ () => undefined }
                placeholderText={ placeholder }
                { ...(isFilterable || onFilter) && { onFilter: (onFilter || this.customFilter) } }
                { ...groups && groups.length > 0 && { isGrouped: true }}
            >
                { groups && groups.length > 0 ? (
                    groups.map(({ value: groupValue, onSelect, label: groupLabel, id: groupId, type, items, ...group }, groupKey) => {
                        const filteredItems = this.mapItems({ groupValue, onSelect, groupLabel, groupId, type, items, ...group }, groupKey);
                        return filteredItems.filter(Boolean).length > 0
                            ? <SelectGroup
                                {...group}
                                key={groupId || groupValue || groupKey}
                                label={groupLabel}
                                id={groupId || `group-${groupValue || groupKey}`}
                            > {filteredItems} </SelectGroup>
                            : <Fragment/>;
                    })
                ) : (
                    this.mapItems({ items }).length > 0 || <Fragment/>
                ) }
            </Select> }
        </Fragment>);
    }
}

const itemsProps = PropTypes.arrayOf(
    PropTypes.shape({
        value: PropTypes.string,
        label: PropTypes.node,
        id: PropTypes.string,
        isChecked: PropTypes.bool,
        onClick: PropTypes.func,
        props: PropTypes.shape({
            [PropTypes.string]: PropTypes.any
        })
    })
);

Group.propTypes = {
    selected: PropTypes.shape({
        [PropTypes.string]: PropTypes.shape({
            [PropTypes.string]: PropTypes.bool
        })
    }),
    onChange: PropTypes.func,
    groups: PropTypes.arrayOf(PropTypes.shape({
        label: PropTypes.node,
        value: PropTypes.string,
        onSelect: PropTypes.func,
        type: PropTypes.oneOf(Object.values(groupType)),
        items: itemsProps
    })),
    items: itemsProps,
    isFilterable: PropTypes.bool,
    onFilter: PropTypes.func
};

Group.defaultProps = {
    selected: {},
    onChange: () => undefined,
    groups: [],
    isFilterable: false
};

export default Group;
