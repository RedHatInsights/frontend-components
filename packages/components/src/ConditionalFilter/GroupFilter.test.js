import React from 'react';
import Group from './GroupFilter';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { act } from 'react-dom/test-utils';

const config = {
    groups: [
        {
            label: 'First value',
            items: [
                {
                    label: 'First value',
                    value: 'first'
                },
                {
                    label: 'Second value'
                }
            ]
        },
        {
            label: 'Second value',
            value: 'second',
            type: 'checkbox',
            items: [
                {
                    label: 'First checkbox'
                },
                {
                    label: 'Second checkbox',
                    value: 'some-value'
                }
            ]
        },
        {
            label: 'Third value',
            value: 'third',
            type: 'radio',
            items: [
                {
                    label: 'First radio'
                },
                {
                    label: 'Second radio'
                }
            ]
        },
        {
            label: 'Groupselectable value',
            groupSelectable: true,
            items: [
                {
                    label: 'First value',
                    value: 'first'
                },
                {
                    label: 'Second value'
                }
            ]
        }
    ]
};

describe('Group - component', () => {
    describe('render', () => {
        it('should render correctly', () => {
            const wrapper = shallow(<Group />);
            expect(toJson(wrapper)).toMatchSnapshot();
        });

        it('should render correctly with items', () => {
            const wrapper = shallow(<Group {...config} />);
            expect(toJson(wrapper)).toMatchSnapshot();
        });

        describe('show more', () => {
            it('should render correctly with items and default text', () => {
                const wrapper = shallow(<Group {...config} onShowMore={() => undefined} />);
                expect(toJson(wrapper)).toMatchSnapshot();
            });

            it('should render correctly with items and custom text', () => {
                const wrapper = shallow(<Group {...config} onShowMore={() => undefined} showMoreTitle="some title" />);
                expect(toJson(wrapper)).toMatchSnapshot();
            });

            it('should render correctly with items and different variant', () => {
                const wrapper = shallow(<Group {...config} onShowMore={() => undefined} showMoreOptions={{
                    variant: 'default'
                }} />);
                expect(toJson(wrapper)).toMatchSnapshot();
            });

            it('should render correctly with items and custom props', () => {
                const wrapper = shallow(<Group {...config} onShowMore={() => undefined} showMoreOptions={{
                    props: {
                        className: 'some-test-class'
                    }
                }} />);
                expect(toJson(wrapper)).toMatchSnapshot();
            });
        });

        it('should render correctly with items - isDisabled', () => {
            const wrapper = shallow(<Group {...config} isDisabled />);
            expect(toJson(wrapper)).toMatchSnapshot();
        });

        it('should render correctly with items and default value', () => {
            const wrapper = shallow(<Group { ...config } value={ [{ value: 'some-value' }] } />);
            expect(toJson(wrapper)).toMatchSnapshot();
        });

        it('should render correctly placeholder', () => {
            const wrapper = shallow(<Group { ...config } placeholder="some placeholder" />);
            expect(toJson(wrapper)).toMatchSnapshot();
        });

        it('should render correctly with items and selected value', () => {
            const currectConfig = { ...config };
            currectConfig.groups[1].items[1].isChecked = true;
            const wrapper = shallow(<Group {...currectConfig} />);
            expect(toJson(wrapper)).toMatchSnapshot();
        });
    });

    describe('API', () => {
        it('should open', () => {
            const wrapper = mount(<Group { ...config } />);
            wrapper.find('button.pf-c-menu-toggle').simulate('click');
            wrapper.update();
            expect(wrapper.find('.pf-c-menu').length > 0).toBe(true);
        });
    });
});
