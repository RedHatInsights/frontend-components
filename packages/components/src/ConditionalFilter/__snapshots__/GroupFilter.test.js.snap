// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Group - component render should render correctly 1`] = `
<Popper
  isVisible={false}
  popper={
    <Menu
      className=""
    >
      <MenuContent>
        <MenuList>
          <span
            hidden={true}
            value=""
          />
        </MenuList>
      </MenuContent>
    </Menu>
  }
  trigger={
    <MenuToggle
      isExpanded={false}
      onClick={[Function]}
    />
  }
/>
`;

exports[`Group - component render should render correctly placeholder 1`] = `
<Popper
  isVisible={false}
  popper={
    <Menu
      className=""
    >
      <MenuContent>
        <MenuList>
          <MenuGroup
            label="First value"
          >
            <MenuItem
              itemId={0}
            >
              First value
            </MenuItem>
            <MenuItem
              itemId={1}
            >
              Second value
            </MenuItem>
          </MenuGroup>
          <MenuGroup
            label="Second value"
          >
            <MenuItem
              itemId={0}
            >
              <Checkbox
                className=""
                id="0"
                isChecked={false}
                isDisabled={false}
                isValid={true}
                label="First checkbox"
                name="0"
                onChange={[Function]}
                onClick={[Function]}
                ouiaSafe={true}
                value="0"
              />
            </MenuItem>
            <MenuItem
              itemId={1}
            >
              <Checkbox
                className=""
                id="some-value"
                isChecked={false}
                isDisabled={false}
                isValid={true}
                label="Second checkbox"
                name="some-value"
                onChange={[Function]}
                onClick={[Function]}
                ouiaSafe={true}
                value="some-value"
              />
            </MenuItem>
          </MenuGroup>
          <MenuGroup
            label="Third value"
          >
            <MenuItem
              itemId={0}
            >
              <Radio
                className=""
                id="0"
                isChecked={false}
                isDisabled={false}
                isValid={true}
                label="First radio"
                name="0"
                onChange={[Function]}
                onClick={[Function]}
                value="0"
              />
            </MenuItem>
            <MenuItem
              itemId={1}
            >
              <Radio
                className=""
                id="1"
                isChecked={false}
                isDisabled={false}
                isValid={true}
                label="Second radio"
                name="1"
                onChange={[Function]}
                onClick={[Function]}
                value="1"
              />
            </MenuItem>
          </MenuGroup>
          <MenuGroup
            label="Groupselectable value"
          >
            <MenuItem
              itemId={0}
            >
              First value
            </MenuItem>
            <MenuItem
              itemId={1}
            >
              Second value
            </MenuItem>
          </MenuGroup>
          <span
            hidden={true}
            value=""
          />
        </MenuList>
      </MenuContent>
    </Menu>
  }
  trigger={
    <MenuToggle
      isExpanded={false}
      onClick={[Function]}
    >
      some placeholder
    </MenuToggle>
  }
/>
`;

exports[`Group - component render should render correctly with items - isDisabled 1`] = `
<Popper
  isVisible={false}
  popper={
    <Menu
      className=""
    >
      <MenuContent>
        <MenuList>
          <MenuGroup
            label="First value"
          >
            <MenuItem
              itemId={0}
            >
              First value
            </MenuItem>
            <MenuItem
              itemId={1}
            >
              Second value
            </MenuItem>
          </MenuGroup>
          <MenuGroup
            label="Second value"
          >
            <MenuItem
              itemId={0}
            >
              <Checkbox
                className=""
                id="0"
                isChecked={false}
                isDisabled={false}
                isValid={true}
                label="First checkbox"
                name="0"
                onChange={[Function]}
                onClick={[Function]}
                ouiaSafe={true}
                value="0"
              />
            </MenuItem>
            <MenuItem
              itemId={1}
            >
              <Checkbox
                className=""
                id="some-value"
                isChecked={false}
                isDisabled={false}
                isValid={true}
                label="Second checkbox"
                name="some-value"
                onChange={[Function]}
                onClick={[Function]}
                ouiaSafe={true}
                value="some-value"
              />
            </MenuItem>
          </MenuGroup>
          <MenuGroup
            label="Third value"
          >
            <MenuItem
              itemId={0}
            >
              <Radio
                className=""
                id="0"
                isChecked={false}
                isDisabled={false}
                isValid={true}
                label="First radio"
                name="0"
                onChange={[Function]}
                onClick={[Function]}
                value="0"
              />
            </MenuItem>
            <MenuItem
              itemId={1}
            >
              <Radio
                className=""
                id="1"
                isChecked={false}
                isDisabled={false}
                isValid={true}
                label="Second radio"
                name="1"
                onChange={[Function]}
                onClick={[Function]}
                value="1"
              />
            </MenuItem>
          </MenuGroup>
          <MenuGroup
            label="Groupselectable value"
          >
            <MenuItem
              itemId={0}
            >
              First value
            </MenuItem>
            <MenuItem
              itemId={1}
            >
              Second value
            </MenuItem>
          </MenuGroup>
          <span
            hidden={true}
            value=""
          />
        </MenuList>
      </MenuContent>
    </Menu>
  }
  trigger={
    <MenuToggle
      isExpanded={false}
      onClick={[Function]}
    />
  }
/>
`;

exports[`Group - component render should render correctly with items 1`] = `
<Popper
  isVisible={false}
  popper={
    <Menu
      className=""
    >
      <MenuContent>
        <MenuList>
          <MenuGroup
            label="First value"
          >
            <MenuItem
              itemId={0}
            >
              First value
            </MenuItem>
            <MenuItem
              itemId={1}
            >
              Second value
            </MenuItem>
          </MenuGroup>
          <MenuGroup
            label="Second value"
          >
            <MenuItem
              itemId={0}
            >
              <Checkbox
                className=""
                id="0"
                isChecked={false}
                isDisabled={false}
                isValid={true}
                label="First checkbox"
                name="0"
                onChange={[Function]}
                onClick={[Function]}
                ouiaSafe={true}
                value="0"
              />
            </MenuItem>
            <MenuItem
              itemId={1}
            >
              <Checkbox
                className=""
                id="some-value"
                isChecked={false}
                isDisabled={false}
                isValid={true}
                label="Second checkbox"
                name="some-value"
                onChange={[Function]}
                onClick={[Function]}
                ouiaSafe={true}
                value="some-value"
              />
            </MenuItem>
          </MenuGroup>
          <MenuGroup
            label="Third value"
          >
            <MenuItem
              itemId={0}
            >
              <Radio
                className=""
                id="0"
                isChecked={false}
                isDisabled={false}
                isValid={true}
                label="First radio"
                name="0"
                onChange={[Function]}
                onClick={[Function]}
                value="0"
              />
            </MenuItem>
            <MenuItem
              itemId={1}
            >
              <Radio
                className=""
                id="1"
                isChecked={false}
                isDisabled={false}
                isValid={true}
                label="Second radio"
                name="1"
                onChange={[Function]}
                onClick={[Function]}
                value="1"
              />
            </MenuItem>
          </MenuGroup>
          <MenuGroup
            label="Groupselectable value"
          >
            <MenuItem
              itemId={0}
            >
              First value
            </MenuItem>
            <MenuItem
              itemId={1}
            >
              Second value
            </MenuItem>
          </MenuGroup>
          <span
            hidden={true}
            value=""
          />
        </MenuList>
      </MenuContent>
    </Menu>
  }
  trigger={
    <MenuToggle
      isExpanded={false}
      onClick={[Function]}
    />
  }
/>
`;

exports[`Group - component render should render correctly with items and default value 1`] = `
<Popper
  isVisible={false}
  popper={
    <Menu
      className=""
    >
      <MenuContent>
        <MenuList>
          <MenuGroup
            label="First value"
          >
            <MenuItem
              itemId={0}
            >
              First value
            </MenuItem>
            <MenuItem
              itemId={1}
            >
              Second value
            </MenuItem>
          </MenuGroup>
          <MenuGroup
            label="Second value"
          >
            <MenuItem
              itemId={0}
            >
              <Checkbox
                className=""
                id="0"
                isChecked={false}
                isDisabled={false}
                isValid={true}
                label="First checkbox"
                name="0"
                onChange={[Function]}
                onClick={[Function]}
                ouiaSafe={true}
                value="0"
              />
            </MenuItem>
            <MenuItem
              itemId={1}
            >
              <Checkbox
                className=""
                id="some-value"
                isChecked={false}
                isDisabled={false}
                isValid={true}
                label="Second checkbox"
                name="some-value"
                onChange={[Function]}
                onClick={[Function]}
                ouiaSafe={true}
                value="some-value"
              />
            </MenuItem>
          </MenuGroup>
          <MenuGroup
            label="Third value"
          >
            <MenuItem
              itemId={0}
            >
              <Radio
                className=""
                id="0"
                isChecked={false}
                isDisabled={false}
                isValid={true}
                label="First radio"
                name="0"
                onChange={[Function]}
                onClick={[Function]}
                value="0"
              />
            </MenuItem>
            <MenuItem
              itemId={1}
            >
              <Radio
                className=""
                id="1"
                isChecked={false}
                isDisabled={false}
                isValid={true}
                label="Second radio"
                name="1"
                onChange={[Function]}
                onClick={[Function]}
                value="1"
              />
            </MenuItem>
          </MenuGroup>
          <MenuGroup
            label="Groupselectable value"
          >
            <MenuItem
              itemId={0}
            >
              First value
            </MenuItem>
            <MenuItem
              itemId={1}
            >
              Second value
            </MenuItem>
          </MenuGroup>
          <span
            hidden={true}
            value=""
          />
        </MenuList>
      </MenuContent>
    </Menu>
  }
  trigger={
    <MenuToggle
      isExpanded={false}
      onClick={[Function]}
    />
  }
/>
`;

exports[`Group - component render should render correctly with items and selected value 1`] = `
<Popper
  isVisible={false}
  popper={
    <Menu
      className=""
    >
      <MenuContent>
        <MenuList>
          <MenuGroup
            label="First value"
          >
            <MenuItem
              itemId={0}
            >
              First value
            </MenuItem>
            <MenuItem
              itemId={1}
            >
              Second value
            </MenuItem>
          </MenuGroup>
          <MenuGroup
            label="Second value"
          >
            <MenuItem
              itemId={0}
            >
              <Checkbox
                className=""
                id="0"
                isChecked={false}
                isDisabled={false}
                isValid={true}
                label="First checkbox"
                name="0"
                onChange={[Function]}
                onClick={[Function]}
                ouiaSafe={true}
                value="0"
              />
            </MenuItem>
            <MenuItem
              itemId={1}
            >
              <Checkbox
                className=""
                id="some-value"
                isChecked={true}
                isDisabled={false}
                isValid={true}
                label="Second checkbox"
                name="some-value"
                onChange={[Function]}
                onClick={[Function]}
                ouiaSafe={true}
                value="some-value"
              />
            </MenuItem>
          </MenuGroup>
          <MenuGroup
            label="Third value"
          >
            <MenuItem
              itemId={0}
            >
              <Radio
                className=""
                id="0"
                isChecked={false}
                isDisabled={false}
                isValid={true}
                label="First radio"
                name="0"
                onChange={[Function]}
                onClick={[Function]}
                value="0"
              />
            </MenuItem>
            <MenuItem
              itemId={1}
            >
              <Radio
                className=""
                id="1"
                isChecked={false}
                isDisabled={false}
                isValid={true}
                label="Second radio"
                name="1"
                onChange={[Function]}
                onClick={[Function]}
                value="1"
              />
            </MenuItem>
          </MenuGroup>
          <MenuGroup
            label="Groupselectable value"
          >
            <MenuItem
              itemId={0}
            >
              First value
            </MenuItem>
            <MenuItem
              itemId={1}
            >
              Second value
            </MenuItem>
          </MenuGroup>
          <span
            hidden={true}
            value=""
          />
        </MenuList>
      </MenuContent>
    </Menu>
  }
  trigger={
    <MenuToggle
      isExpanded={false}
      onClick={[Function]}
    />
  }
/>
`;

exports[`Group - component render show more should render correctly with items and custom props 1`] = `
<Popper
  isVisible={false}
  popper={
    <Menu
      className=""
    >
      <MenuContent>
        <MenuList>
          <MenuGroup
            label="First value"
          >
            <MenuItem
              itemId={0}
            >
              First value
            </MenuItem>
            <MenuItem
              itemId={1}
            >
              Second value
            </MenuItem>
          </MenuGroup>
          <MenuGroup
            label="Second value"
          >
            <MenuItem
              itemId={0}
            >
              <Checkbox
                className=""
                id="0"
                isChecked={false}
                isDisabled={false}
                isValid={true}
                label="First checkbox"
                name="0"
                onChange={[Function]}
                onClick={[Function]}
                ouiaSafe={true}
                value="0"
              />
            </MenuItem>
            <MenuItem
              itemId={1}
            >
              <Checkbox
                className=""
                id="some-value"
                isChecked={false}
                isDisabled={false}
                isValid={true}
                label="Second checkbox"
                name="some-value"
                onChange={[Function]}
                onClick={[Function]}
                ouiaSafe={true}
                value="some-value"
              />
            </MenuItem>
          </MenuGroup>
          <MenuGroup
            label="Third value"
          >
            <MenuItem
              itemId={0}
            >
              <Radio
                className=""
                id="0"
                isChecked={false}
                isDisabled={false}
                isValid={true}
                label="First radio"
                name="0"
                onChange={[Function]}
                onClick={[Function]}
                value="0"
              />
            </MenuItem>
            <MenuItem
              itemId={1}
            >
              <Radio
                className=""
                id="1"
                isChecked={false}
                isDisabled={false}
                isValid={true}
                label="Second radio"
                name="1"
                onChange={[Function]}
                onClick={[Function]}
                value="1"
              />
            </MenuItem>
          </MenuGroup>
          <MenuGroup
            label="Groupselectable value"
          >
            <MenuItem
              itemId={0}
            >
              First value
            </MenuItem>
            <MenuItem
              itemId={1}
            >
              Second value
            </MenuItem>
          </MenuGroup>
          <MenuItem
            className="ins-c-menu__show--more"
            itemId="loader"
            onClick={[Function]}
            props={
              Object {
                "className": "some-test-class",
              }
            }
          />
        </MenuList>
      </MenuContent>
    </Menu>
  }
  trigger={
    <MenuToggle
      isExpanded={false}
      onClick={[Function]}
    />
  }
/>
`;

exports[`Group - component render show more should render correctly with items and custom text 1`] = `
<Popper
  isVisible={false}
  popper={
    <Menu
      className=""
    >
      <MenuContent>
        <MenuList>
          <MenuGroup
            label="First value"
          >
            <MenuItem
              itemId={0}
            >
              First value
            </MenuItem>
            <MenuItem
              itemId={1}
            >
              Second value
            </MenuItem>
          </MenuGroup>
          <MenuGroup
            label="Second value"
          >
            <MenuItem
              itemId={0}
            >
              <Checkbox
                className=""
                id="0"
                isChecked={false}
                isDisabled={false}
                isValid={true}
                label="First checkbox"
                name="0"
                onChange={[Function]}
                onClick={[Function]}
                ouiaSafe={true}
                value="0"
              />
            </MenuItem>
            <MenuItem
              itemId={1}
            >
              <Checkbox
                className=""
                id="some-value"
                isChecked={false}
                isDisabled={false}
                isValid={true}
                label="Second checkbox"
                name="some-value"
                onChange={[Function]}
                onClick={[Function]}
                ouiaSafe={true}
                value="some-value"
              />
            </MenuItem>
          </MenuGroup>
          <MenuGroup
            label="Third value"
          >
            <MenuItem
              itemId={0}
            >
              <Radio
                className=""
                id="0"
                isChecked={false}
                isDisabled={false}
                isValid={true}
                label="First radio"
                name="0"
                onChange={[Function]}
                onClick={[Function]}
                value="0"
              />
            </MenuItem>
            <MenuItem
              itemId={1}
            >
              <Radio
                className=""
                id="1"
                isChecked={false}
                isDisabled={false}
                isValid={true}
                label="Second radio"
                name="1"
                onChange={[Function]}
                onClick={[Function]}
                value="1"
              />
            </MenuItem>
          </MenuGroup>
          <MenuGroup
            label="Groupselectable value"
          >
            <MenuItem
              itemId={0}
            >
              First value
            </MenuItem>
            <MenuItem
              itemId={1}
            >
              Second value
            </MenuItem>
          </MenuGroup>
          <MenuItem
            className="ins-c-menu__show--more"
            itemId="loader"
            onClick={[Function]}
          >
            some title
          </MenuItem>
        </MenuList>
      </MenuContent>
    </Menu>
  }
  trigger={
    <MenuToggle
      isExpanded={false}
      onClick={[Function]}
    />
  }
/>
`;

exports[`Group - component render show more should render correctly with items and default text 1`] = `
<Popper
  isVisible={false}
  popper={
    <Menu
      className=""
    >
      <MenuContent>
        <MenuList>
          <MenuGroup
            label="First value"
          >
            <MenuItem
              itemId={0}
            >
              First value
            </MenuItem>
            <MenuItem
              itemId={1}
            >
              Second value
            </MenuItem>
          </MenuGroup>
          <MenuGroup
            label="Second value"
          >
            <MenuItem
              itemId={0}
            >
              <Checkbox
                className=""
                id="0"
                isChecked={false}
                isDisabled={false}
                isValid={true}
                label="First checkbox"
                name="0"
                onChange={[Function]}
                onClick={[Function]}
                ouiaSafe={true}
                value="0"
              />
            </MenuItem>
            <MenuItem
              itemId={1}
            >
              <Checkbox
                className=""
                id="some-value"
                isChecked={false}
                isDisabled={false}
                isValid={true}
                label="Second checkbox"
                name="some-value"
                onChange={[Function]}
                onClick={[Function]}
                ouiaSafe={true}
                value="some-value"
              />
            </MenuItem>
          </MenuGroup>
          <MenuGroup
            label="Third value"
          >
            <MenuItem
              itemId={0}
            >
              <Radio
                className=""
                id="0"
                isChecked={false}
                isDisabled={false}
                isValid={true}
                label="First radio"
                name="0"
                onChange={[Function]}
                onClick={[Function]}
                value="0"
              />
            </MenuItem>
            <MenuItem
              itemId={1}
            >
              <Radio
                className=""
                id="1"
                isChecked={false}
                isDisabled={false}
                isValid={true}
                label="Second radio"
                name="1"
                onChange={[Function]}
                onClick={[Function]}
                value="1"
              />
            </MenuItem>
          </MenuGroup>
          <MenuGroup
            label="Groupselectable value"
          >
            <MenuItem
              itemId={0}
            >
              First value
            </MenuItem>
            <MenuItem
              itemId={1}
            >
              Second value
            </MenuItem>
          </MenuGroup>
          <MenuItem
            className="ins-c-menu__show--more"
            itemId="loader"
            onClick={[Function]}
          />
        </MenuList>
      </MenuContent>
    </Menu>
  }
  trigger={
    <MenuToggle
      isExpanded={false}
      onClick={[Function]}
    />
  }
/>
`;

exports[`Group - component render show more should render correctly with items and different variant 1`] = `
<Popper
  isVisible={false}
  popper={
    <Menu
      className=""
    >
      <MenuContent>
        <MenuList>
          <MenuGroup
            label="First value"
          >
            <MenuItem
              itemId={0}
            >
              First value
            </MenuItem>
            <MenuItem
              itemId={1}
            >
              Second value
            </MenuItem>
          </MenuGroup>
          <MenuGroup
            label="Second value"
          >
            <MenuItem
              itemId={0}
            >
              <Checkbox
                className=""
                id="0"
                isChecked={false}
                isDisabled={false}
                isValid={true}
                label="First checkbox"
                name="0"
                onChange={[Function]}
                onClick={[Function]}
                ouiaSafe={true}
                value="0"
              />
            </MenuItem>
            <MenuItem
              itemId={1}
            >
              <Checkbox
                className=""
                id="some-value"
                isChecked={false}
                isDisabled={false}
                isValid={true}
                label="Second checkbox"
                name="some-value"
                onChange={[Function]}
                onClick={[Function]}
                ouiaSafe={true}
                value="some-value"
              />
            </MenuItem>
          </MenuGroup>
          <MenuGroup
            label="Third value"
          >
            <MenuItem
              itemId={0}
            >
              <Radio
                className=""
                id="0"
                isChecked={false}
                isDisabled={false}
                isValid={true}
                label="First radio"
                name="0"
                onChange={[Function]}
                onClick={[Function]}
                value="0"
              />
            </MenuItem>
            <MenuItem
              itemId={1}
            >
              <Radio
                className=""
                id="1"
                isChecked={false}
                isDisabled={false}
                isValid={true}
                label="Second radio"
                name="1"
                onChange={[Function]}
                onClick={[Function]}
                value="1"
              />
            </MenuItem>
          </MenuGroup>
          <MenuGroup
            label="Groupselectable value"
          >
            <MenuItem
              itemId={0}
            >
              First value
            </MenuItem>
            <MenuItem
              itemId={1}
            >
              Second value
            </MenuItem>
          </MenuGroup>
          <MenuItem
            className="ins-c-menu__show--more"
            itemId="loader"
            onClick={[Function]}
            variant="default"
          />
        </MenuList>
      </MenuContent>
    </Menu>
  }
  trigger={
    <MenuToggle
      isExpanded={false}
      onClick={[Function]}
    />
  }
/>
`;
